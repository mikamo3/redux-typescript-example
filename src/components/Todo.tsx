import React from "react";
import { types } from "../state/ducks/todo";
import { Text } from "react-native";

interface Props {
  todo: types.Todo;
  onPress: () => void;
}

const Todo: React.FC<Props> = ({ todo, onPress }) => (
  <Text
    style={{ textDecorationLine: todo.completed ? "line-through" : "none" }}
    onPress={onPress}
  >
    {todo.text}
  </Text>
);
export default Todo;
