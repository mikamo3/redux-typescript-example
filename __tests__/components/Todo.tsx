import React from "react";
import { Text } from "react-native";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import Todo from "../../src/components/Todo";

describe("snapshot", () => {
  it("correct", () => {
    const TodoComponent = renderer.create(
      <Todo completed={true} text="hoge" onPress={() => {}} />
    );
    expect(TodoComponent).toMatchSnapshot();
  });
});

describe("props", () => {
  describe("completedがtrueの場合", () => {
    it('TextのstyleのtextDecorationLineが"line-through"であること', () => {
      const TodoComponent = renderer.create(
        <Todo completed={true} text="hoge" onPress={() => {}} />
      );
      const text = TodoComponent.root.findByType(Text);
      expect(text.props.style).toMatchObject({
        textDecorationLine: "line-through"
      });
    });
  });
  describe("completedがfalseの場合", () => {
    it('TextのstyleのtextDecorationLineが"none"であること', () => {
      const TodoComponent = renderer.create(
        <Todo completed={false} text="hoge" onPress={() => {}} />
      );
      const text = TodoComponent.root.findByType(Text);
      expect(text.props.style).toMatchObject({
        textDecorationLine: "none"
      });
    });
  });
  describe("text", () => {
    it("子要素に指定したtextが存在すること", () => {
      const TodoComponent = renderer.create(
        <Todo completed={false} text="hoge" onPress={() => {}} />
      );
      const text = TodoComponent.root.findByType(Text);
      expect(text.props.children).toEqual("hoge");
    });
  });
});

describe("イベント", () => {
  describe("Textが押下された時", () => {
    it("onPressが呼び出されること", () => {
      const onPress = jest.fn();
      const todoComponent = shallow(
        <Todo completed={false} text="hoge" onPress={onPress} />
      );
      const text = todoComponent.find(Text);
      text.simulate("press");
      expect(onPress.mock.calls).toHaveLength(1);
    });
  });
});
