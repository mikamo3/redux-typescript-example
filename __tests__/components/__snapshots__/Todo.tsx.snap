// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot correct 1`] = `
<Text
  onPress={[Function]}
  style={
    Object {
      "textDecorationLine": "line-through",
    }
  }
>
  hoge
</Text>
`;
